core/state/statedb.go

```diff
@@ -159,10 +162,10 @@ func NewWithSnapLayers(root common.Hash, db Database, snaps *snapshot.Tree, laye
 		}
 	}
 
-	if substate.RecordReplay {
+	if rr.RecordReplay {
```

Comment:
It seems odd that this flag deciding whether recording is enabled or not is coming from the substate module. Should it be defined in the go-ethereum-substate module instead?

Reply:
The thing is this global is for some reason used in many packages and cannot be removed from `Substate` because this repo is only one used in all of these packages. So we havent found a quick solution for this.
Nevertheless, this global is a big problem and we are currently trying to find a way to get rid of it.

Reply:
I talked to Wasuwee about it, let's wait for a review from her.